### PR TITLE
fix: ensure correct element is activated when label is clicked

### PR DESCRIPTION
**Pull request by danielstorey**

### CODE REVIEW

File: src/components/Checkbox/Checkbox.vue [hunk 1]
```diff
@@ -87,4 +87,10 @@ const checkIconClass = computed(() => {
 
 useRegisterInput(props, el);
 
+// The Switch component does not accept an id prop
```
Comment (robiharid) [inline src/components/Checkbox/Checkbox.vue:90]: typo: I think you meant Checkbox component here

File: src/components/Checkbox/Checkbox.vue [hunk 2]
```diff
@@ -87,4 +87,10 @@ const checkIconClass = computed(() => {
...
+// so we need to override headless UI's default id
+// after the component is mounted
+onMounted(() => {
+  switchRef.value.el.id = props.id;
```
Comment (robiharid) [inline src/components/Checkbox/Checkbox.vue:94]: Is it better to have the id as a different prop, so you don't have to keep overriding Headless UI's ids?
i.e. `<Switch C-id="id">`
(or perhaps the other components need the :id directly, not sure, just a suggestion)

**Review by robiharid — APPROVED** [review]

File: src/components/Checkbox/Checkbox.vue (hunk 1 above)
Comment (danielstorey) [reply src/components/Checkbox/Checkbox.vue:90]: No, I meant headless UI's switch component :)

File: src/components/Checkbox/Checkbox.vue (hunk 2 above)
Comment (danielstorey) [reply src/components/Checkbox/Checkbox.vue:94]: If we did that we'd have to do some extra work to handle what happens when the label is clicked. This way the label gets bound to the right element. I don't think we need to worry about side effects as we're not using headless UI's label components

File: src/components/Checkbox/Checkbox.vue (hunk 1 above)
Comment (robiharid) [reply src/components/Checkbox/Checkbox.vue:90]: Must be seeing things (on my end)!

**Review by CalumH0 — APPROVED** [review]

**Pull request merged**